crates/cosmos/cosmos-test-components/src/chain/impls/recover_client.rs

```diff
@@ -0,0 +1,148 @@
+use core::marker::PhantomData;
+use core::time::Duration;
+
+use hermes_core::chain_components::traits::{
+    CanQueryClientStatus, HasAmountType, HasClientIdType, HasClientStatusMethods, HasDenomType,
+};
+use hermes_core::logging_components::traits::CanLogMessage;
+use hermes_core::relayer_components::multi::traits::chain_at::HasChainTypeAt;
+use hermes_core::relayer_components::transaction::traits::CanSendMessagesWithSigner;
+use hermes_prelude::*;
+use hermes_test_components::chain::traits::{
+    CanBuildDepositProposalMessage, CanBuildVoteProposalMessage, CanQueryProposalStatus,
+    HasWalletSigner,
+};
+use hermes_test_components::chain::types::{ProposalStatus, ProposalVote};
+use hermes_test_components::chain_driver::traits::{
+    CanGenerateRandomAmount, HasChain, HasDenom, HasWallet, StakingDenom, ValidatorWallet,
+};
+use hermes_test_components::driver::traits::HasChainDriverAt;
+use hermes_test_components::test_case::traits::recover_client::{
+    RecoverClientHandler, RecoverClientHandlerComponent,
+};
+use ibc::core::host::types::identifiers::ClientId;
+
+const MAX_RETRIES_FOR_PROPOSAL_STATUS: usize = 15;
+const WAIT_SECONDS_FOR_PROPOSAL_STATUS: u64 = 1;
```

Comment:
Just wanted to confirm that these parameters are defined here as constants like this -- as they are part of a test component.

If it was a proper chain component, we should have probably create getter methods for these parameters, right ?

Reply:
Yes, this is a faster solution but more rigid. Since it is specifically, and currently only, for the test I opted for the solution using constants.